src/ImageSharp.Web/Providers/FileProviderImageProvider.cs

```diff
@@ -30,20 +30,23 @@ public class FileProviderImageProvider : IImageProvider
         /// </summary>
         /// <param name="fileProvider">The file provider.</param>
         /// <param name="formatUtilities">Contains various format helper methods based on the current configuration.</param>
-        public FileProviderImageProvider(IFileProvider fileProvider, FormatUtilities formatUtilities)
+        protected FileProviderImageProvider(IFileProvider fileProvider, FormatUtilities formatUtilities)
         {
-            this.fileProvider = fileProvider ?? throw new ArgumentNullException(nameof(fileProvider));
-            this.formatUtilities = formatUtilities ?? throw new ArgumentNullException(nameof(formatUtilities));
+            Guard.NotNull(fileProvider, nameof(fileProvider));
+            Guard.NotNull(formatUtilities, nameof(formatUtilities));
+
+            this.fileProvider = fileProvider;
+            this.formatUtilities = formatUtilities;
         }
 
         /// <inheritdoc/>
-        public ProcessingBehavior ProcessingBehavior { get; protected set; } = ProcessingBehavior.CommandOnly;
+        public virtual ProcessingBehavior ProcessingBehavior { get; protected set; } = ProcessingBehavior.CommandOnly;
```

Comment:
Any reason for making this `virtual` _and_ adding the protected setter? This will add the requirement to include the setter if you're overriding this property...

Reply:
We need to be able to make this property configurable.